File: src/main/java/me/chell/samsara/impl/module/movement/PullDown.java
package me.chell.samsara.impl.module.movement;

import me.chell.samsara.api.event.PlayerUpdateEvent;
import me.chell.samsara.api.module.Module;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

public class PullDown extends Module {
    public PullDown() {
        super("PullDown", Category.MOVEMENT);
    }

    @SubscribeEvent
    public void onPlayerUpdate(PlayerUpdateEvent event) {
        if (getPlayer().onGround && getPlayer().isOffsetPositionInLiquid(0D, 0D, 0D)) // no the liquid check shouldn't be inverted
            --getPlayer().motionY;
    }
}

File: src/main/kotlin/me/chell/samsara/impl/module/movement/PullDown.kt
package me.chell.samsara.impl.module.movement

import me.chell.samsara.api.event.EventHandler
import me.chell.samsara.api.feature.Module
import me.chell.samsara.impl.event.PlayerTickEvent
import net.minecraft.tag.FluidTags

class PullDown: Module("PullDown", Category.MOVEMENT, "Makes you fall faster.") {

    @EventHandler
    fun onPlayerTick(event: PlayerTickEvent) {
        if(player.isOnGround && player.getFluidHeight(FluidTags.WATER) == 0.0 && player.getFluidHeight(FluidTags.LAVA) == 0.0)
                player.addVelocity(0.0, -1.0, 0.0)
    }

}